mitodl/micromasters
# Implemented redesign of dashboard page

#### What are the relevant tickets?
Fixes #766 

#### What's this PR do?
Changes dashboard UI code to match mockups on #766, with the exception of certain cases mentioned in #833 

#### Where should the reviewer start?
static/js/components/dashboard/CourseListCard.js

#### How should this be manually tested?
Get a bunch of interesting runs and courses to display in your dashboard, or stub out `getDashboard` in `api.js` like this:
    import { DASHBOARD_RESPONSE } from '../constants';
    export function getDashboard() {
      return Promise.resolve(DASHBOARD_RESPONSE);
    }

Look carefully at the mockups and the course descriptions and verify that all cases match up as expected.

#### Any background context you want to provide?
All programs are shown in this PR, one per card. It looks like the mockup is for a single program but I'm not sure where the user specifies this.

#### Screenshots (if appropriate)

![screenshot from 2016-08-11 17-47-02](https://cloud.githubusercontent.com/assets/863262/17606092/a6d78122-5feb-11e6-8962-b6dfe781a710.png)
![screenshot from 2016-08-11 17-47-11](https://cloud.githubusercontent.com/assets/863262/17606091/a6d72d4e-5feb-11e6-95a8-c5884d88ce84.png)
